scootapi/server/job_status.go

```diff
@@ -87,3 +92,24 @@ func convertSagaStateToJobStatus(sagaState *s.SagaState) *scoot.JobStatus {
 
 	return js
 }
+
+func setTaskData(js *scoot.JobStatus, id string, resultsFromSaga []byte) error {
+	workerRunStatus := worker.RunStatus{}
+	thrifthelpers.JsonDeserialize(&workerRunStatus, resultsFromSaga)
+
+	scootRunStatus := scoot.RunStatus{}
+	var err error
+	scootRunStatus.Status, err = scoot.RunStatusStateFromString(workerRunStatus.Status.String())
+	if err != nil {
+		return err
+	}
+
+	scootRunStatus.RunId = fmt.Sprintf("%s", workerRunStatus.RunId)
+	scootRunStatus.OutUri = workerRunStatus.OutUri
+	scootRunStatus.ErrUri = workerRunStatus.ErrUri
+	scootRunStatus.ExitCode = workerRunStatus.ExitCode
+	scootRunStatus.Error = workerRunStatus.Error
+
+	js.TaskData[id] = &scootRunStatus
```

Comment:
This looks like another Thrift <-> Domain translation layer, no?

I think it's easiest if those return values instead of setting. WDYT?